### DISCUSSION TITLE
Is there a way to know if an overlay is active?

### DISCUSSION THREAD

**Discussion by warpdesign** [palantir/blueprint#6006] [Q&A]
I know I could check that body has the `bp4-overlay-open` but is there another way that doesn't involve poking DOM elements?

**Comment by adidahiya** [comment]
@warpdesign are you looking for an imperative API to tell if _any Blueprint overlay_ is open in the document (dialog, popover, tooltip, drawer, context menu, etc...)? Something like the following?

```ts
import { Overlay } from "@blueprintjs/core";

Overlay.isAnyOverlayOpen();
```

Technically this might be possible, since we do keep track of the overlay stack in this component:

https://github.com/palantir/blueprint/blob/c3f267f256771295bf21466320c1a20e6ae57ba0/packages/core/src/components/overlay/overlay.tsx#L232

But it may be error prone in the rare case when there are multiple copies of @blueprintjs/core on the page (which is not encouraged anyway).

What are you hoping to do with this information, anyway?

**Comment by warpdesign** [comment]
I am working on a [file manager](https://github.com/warpdesign/react-explorer) app using Electron and need to disable some native menus when an overlay is opened. Since I need to be notified when an overlay is opened/closed, such an API wouldn't be enough.

I decided to add an observer using the MutationObserver API: that does the trick.
